File: lab_3_3.cpp
#include <iostream>
#include <cmath>
#include <numeric>
#include <limits>

using namespace std;

// Функция для проверки, является ли число рациональным
bool is_rational(double sum, long long max_denominator = 1000) {
    for (long long denom = 1; denom <= max_denominator; ++denom) {
        double numerator = round(sum * denom);
        if (fabs(numerator / denom - sum) < 1e-9) {
            return true;
        }
    }
    return false;
}

// Функция для вычисления НОД
long long gcd(long long a, long long b) {
    while (b != 0) {
        long long temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

// Функция для представления числа в виде дроби
void to_fraction(double sum, long long &numerator, long long &denominator, long long max_denominator = 1000) {
    for (denominator = 1; denominator <= max_denominator; ++denominator) {
        numerator = round(sum * denominator);
        if (fabs(static_cast<double>(numerator) / denominator - sum) < 1e-9) {
            long long common_divisor = gcd(abs(numerator), denominator);
            numerator /= common_divisor;
            denominator /= common_divisor;
            return;
        }
    }
    numerator = round(sum);
    denominator = 1;
}

int main() {
    int a, b;
    cin >> a >> b;

    // Проверка сходимости по признаку Даламбера
    if (b <= 1) {
        cout << "infinity" << endl;
        return 0;
    }

    // Численное приближение суммы ряда
    double sum = 0.0;
    double epsilon = 1e-10; // Точность
    double term;
    int n = 1;
    do {
        term = pow(n, a) / pow(b, n);
        sum += term;
        n++;
    } while (term > epsilon);

    // Проверка на рациональность
    if (is_rational(sum)) {
        long long numerator, denominator;
        to_fraction(sum, numerator, denominator);
        cout << numerator << "/" << denominator << endl;
    } else {
        cout << "irrational" << endl;
    }

    return 0;
}



// краткое 

File: lab_3_3.rb
# Функция для проверки, является ли число рациональным
def is_rational(sum, max_denominator = 1000)
  (1..max_denominator).each do |denom|
    numerator = (sum * denom).round
    if (numerator.to_f / denom - sum).abs < 1e-9
      return true
    end
  end
  false
end

# Функция для вычисления НОД
def gcd(a, b)
  while b != 0
    a, b = b, a % b
  end
  a
end

# Функция для представления числа в виде дроби
def to_fraction(sum, max_denominator = 1000)
  numerator = 0
  denominator = 1
  
  (1..max_denominator).each do |denom|
    num = (sum * denom).round
    if (num.to_f / denom - sum).abs < 1e-9
      common_divisor = gcd(num.abs, denom)
      numerator = num / common_divisor
      denominator = denom / common_divisor
      return [numerator, denominator]
    end
  end
  
  [(sum.round), 1]
end

# Основная программа
a, b = gets.split.map(&:to_i)

# Проверка сходимости по признаку Даламбера
if b <= 1
  puts "infinity"
  exit
end

# Численное приближение суммы ряда
sum = 0.0
epsilon = 1e-10 # Точность
term = 0.0
n = 1

loop do
  term = (n**a).to_f / (b**n)
  sum += term
  n += 1
  break if term <= epsilon
end

# Проверка на рациональность и вывод результата
if is_rational(sum)
  numerator, denominator = to_fraction(sum)
  puts "#{numerator}/#{denominator}"
else
  puts "irrational"
end
